bottlecap/src/lifecycle/invocation/triggers/event_bridge_event.rs

```diff
@@ -0,0 +1,176 @@
+use chrono::{DateTime, Utc};
+use datadog_trace_protobuf::pb::Span;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use tracing::debug;
+
+use crate::lifecycle::invocation::{
+    processor::S_TO_NS,
+    triggers::{Trigger, DATADOG_CARRIER_KEY, FUNCTION_TRIGGER_EVENT_SOURCE_TAG},
+};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct EventBridgeEvent {
+    pub id: String,
+    pub version: String,
+    pub account: String,
+    pub time: DateTime<Utc>,
+    pub region: String,
+    pub resources: Vec<String>,
+    pub source: String,
+    #[serde(rename = "detail-type")]
+    pub detail_type: String,
+    pub detail: Value,
+    #[serde(rename = "replay-name")]
+    pub replay_name: Option<String>,
+}
+
+impl Trigger for EventBridgeEvent {
+    fn new(payload: Value) -> Option<Self> {
+        match serde_json::from_value(payload) {
+            Ok(event) => Some(event),
+            Err(e) => {
+                debug!("Failed to deserialize EventBridge Event: {}", e);
+                None
+            }
+        }
+    }
+
+    fn is_match(payload: &Value) -> bool {
+        payload.get("detail-type").is_some()
+            && payload
+                .get("source")
+                .and_then(Value::as_str)
+                .map_or(false, |s| s != "aws.events")
+    }
+
+    #[allow(clippy::cast_possible_truncation)]
+    fn enrich_span(&self, span: &mut Span) {
+        // EventBridge events have a timestamp resolution in seconds
+        let start_time = self
+            .time
+            .timestamp_nanos_opt()
+            .unwrap_or((self.time.timestamp_millis() as f64 * S_TO_NS) as i64);
+
+        // todo: service mapping and peer service
+        let service_name = "eventbridge";
+
+        span.name = String::from("aws.eventbridge");
+        span.service = service_name.to_string();
+        span.resource.clone_from(&self.source);
+        span.r#type = String::from("web");
+        span.start = start_time;
+        span.meta.extend(HashMap::from([
+            ("operation_name".to_string(), "aws.eventbridge".to_string()),
+            ("detail_type".to_string(), self.detail_type.clone()),
+        ]));
+    }
+
+    fn get_tags(&self) -> HashMap<String, String> {
+        HashMap::from([(
+            FUNCTION_TRIGGER_EVENT_SOURCE_TAG.to_string(),
+            "eventbridge".to_string(),
+        )])
+    }
+
+    fn get_arn(&self, _region: &str) -> String {
+        self.source.clone()
+    }
+
+    fn get_carrier(&self) -> HashMap<String, String> {
+        if let Ok(detail) = serde_json::from_value::<HashMap<String, Value>>(self.detail.clone()) {
+            if let Some(carrier) = detail.get(DATADOG_CARRIER_KEY) {
+                return serde_json::from_value::<HashMap<String, String>>(carrier.clone())
+                    .unwrap_or_default();
+            }
```

Comment:
Updated the carrier code here, I really liked how we expect the detail to be a `Value`, might be really important to do this in every other field which could be a carrier